From 29f95ff9ab8ce08e59022308be8102f0a4721669 Mon Sep 17 00:00:00 2001
From: Uri Mashiach <uri.mashiach@compulab.co.il>
Date: Thu, 3 Dec 2020 17:44:00 +0200
Subject: [PATCH 58/89] iot-gate-imx8: add support for the IE-POE board

Power Over Ethernet board features:
* 10/100Base-T port.
* Provide up to 12W

Signed-off-by: Uri Mashiach <uri.mashiach@compulab.co.il>
---
 arch/arm64/boot/dts/compulab/Makefile         |  1 +
 .../boot/dts/compulab/sb-iotgimx8-poed.dts    | 80 +++++++++++++++++++
 arch/arm64/configs/iot-gate-imx8_defconfig    |  1 +
 3 files changed, 82 insertions(+)
 create mode 100644 arch/arm64/boot/dts/compulab/sb-iotgimx8-poed.dts

diff --git a/arch/arm64/boot/dts/compulab/Makefile b/arch/arm64/boot/dts/compulab/Makefile
index fa5662fdb782..89d834f6c7c3 100644
--- a/arch/arm64/boot/dts/compulab/Makefile
+++ b/arch/arm64/boot/dts/compulab/Makefile
@@ -11,6 +11,7 @@ dtb-$(CONFIG_ARCH_MXC) += ucm-imx8m-mini.dtb
 dtb-$(CONFIG_ARCH_MXC) += sb-iotgimx8.dtb
 dtb-$(CONFIG_ARCH_MXC) += sb-iotgimx8-can.dtb
 dtb-$(CONFIG_ARCH_MXC) += sb-iotgimx8-ied.dtb
+dtb-$(CONFIG_ARCH_MXC) += sb-iotgimx8-poed.dtb
 
 always-y         := $(dtb-y)
 subdir-y       := $(dts-dirs)
diff --git a/arch/arm64/boot/dts/compulab/sb-iotgimx8-poed.dts b/arch/arm64/boot/dts/compulab/sb-iotgimx8-poed.dts
new file mode 100644
index 000000000000..6159170522da
--- /dev/null
+++ b/arch/arm64/boot/dts/compulab/sb-iotgimx8-poed.dts
@@ -0,0 +1,80 @@
+/*
+ * Copyright (C) 2020 CompuLab Ltd.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ */
+
+#include "sb-iotgimx8.dts"
+
+/ {
+	model = "CompuLab SB-IOTGIMX8 board + IE-POED extension";
+	compatible = "sb-iotgimx8-poed", "sb-iotgimx8", "cpl,ucm-imx8m-mini", "fsl,imx8mm";
+
+};
+
+&i2c4 {
+	clock-frequency = <100000>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_i2c4>;
+	status = "okay";
+
+	eeprom@54 {
+		compatible = "atmel,24c08";
+		reg = <0x54>;
+		pagesize = <16>;
+	};
+};
+
+&ecspi2 {
+	/delete-property/ dmas;
+	/delete-property/ dma-names;
+	#address-cells = <1>;
+	#size-cells = <0>;
+	fsl,spi-num-chipselects = <1>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_ecspi2 &pinctrl_ecspi2_cs>;
+	cs-gpios = <&gpio5 13 GPIO_ACTIVE_LOW>;
+	status = "okay";
+
+	encx24j600@0 {
+		compatible = "microchip,encx24j600";
+		spi-max-frequency = <40000000>;
+		reg = <0>;
+		interrupt-parent = <&gpio3>;
+		interrupts = <1 IRQ_TYPE_EDGE_FALLING>;
+	};
+};
+
+&iomuxc {
+	sb-iotgimx8-poed {
+		pinctrl_i2c4: i2c4grp {
+			fsl,pins = <
+				MX8MM_IOMUXC_I2C4_SCL_I2C4_SCL		0x400001c3
+				MX8MM_IOMUXC_I2C4_SDA_I2C4_SDA		0x400001c3
+			>;
+		};
+
+		pinctrl_ecspi2: ecspi2grp {
+			fsl,pins = <
+				MX8MM_IOMUXC_ECSPI2_SCLK_ECSPI2_SCLK	0x02
+				MX8MM_IOMUXC_ECSPI2_MOSI_ECSPI2_MOSI	0x02
+				MX8MM_IOMUXC_ECSPI2_MISO_ECSPI2_MISO	0x102
+			>;
+		};
+
+		pinctrl_ecspi2_cs: ecspi2_csgrp {
+			fsl,pins = <
+				MX8MM_IOMUXC_ECSPI2_SS0_GPIO5_IO13	0x4000
+				MX8MM_IOMUXC_NAND_CE0_B_GPIO3_IO1	0x04146
+			>;
+		};
+	};
+};
diff --git a/arch/arm64/configs/iot-gate-imx8_defconfig b/arch/arm64/configs/iot-gate-imx8_defconfig
index c6c91a436a76..912ac31d0093 100644
--- a/arch/arm64/configs/iot-gate-imx8_defconfig
+++ b/arch/arm64/configs/iot-gate-imx8_defconfig
@@ -239,6 +239,7 @@ CONFIG_SKY2=y
 CONFIG_MLX4_EN=m
 CONFIG_MLX5_CORE=m
 CONFIG_MLX5_CORE_EN=y
+CONFIG_ENCX24J600=m
 CONFIG_MSCC_OCELOT_SWITCH=y
 CONFIG_QCOM_EMAC=m
 CONFIG_SMC91X=y
-- 
2.17.1

